doc/user/content/ingest-data/mysql/self-hosted.md

```diff
@@ -0,0 +1,472 @@
+---
+title: "Ingest data from self-hosted MySQL"
+description: "How to stream data from self-hosted MySQL database to Materialize"
+draft: true
+menu:
+  main:
+    parent: "MySQL"
+    name: "Self-hosted"
+    weight: 25
+---
+
+This page shows you how to stream data from a self-hosted MySQL database to
+Materialize using the [MySQL source](/sql/create-source/mysql/).
+
+## Before you begin
+
+{{% mysql-direct/before-you-begin %}}
+
+## Step 1. Enable GTID-based replication
+
+Materialize uses MySQL's [logical replication](https://www.MySQL.org/docs/current/logical-replication.html)
+protocol to track changes in your database and propagate them to Materialize.
+
+As a first step, you need to make sure logical replication is enabled.
+
+1. As a _superuser_, use `psql` (or your preferred SQL client) to connect to your
+   MySQL database.
+
+1. Check if logical replication is enabled:
+
+    ```sql
+    SHOW wal_level;
+    ```
+
+    The `wal_level` setting must be set to `logical`. If it's set to any other
+    value, change it to `logical` in the database configuration file
+    (`MySQL.conf`).
+
+1. Restart the database in order for the new `wal_level` to take effect. Keep in
+   mind that restarting can affect database performance.
+
+1. Back in the SQL client connected to MySQL, verify that replication is
+   now enabled:
+
+    ```sql
+    SHOW wal_level;
+    ```
```

Comment:
this is postgres

Reply:
Yup. This one is marked as draft because we don't have the sequence of steps. Will add the self-hosted guide as a follow-up.

Reply:
Draft as in, it won't show up in the actual user-facing documentation.

Reply:
Ended up adding it in.